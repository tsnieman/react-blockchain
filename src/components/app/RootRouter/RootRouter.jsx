import React from 'react';

// Routing
import { BrowserRouter, Match, Miss } from 'react-router';

// Declarative webpack code-splitting using Webpack 2 + System.import
import LazyRoute from 'lazy-route';

import Header from 'components/app/Header';

const RootRouter = () => (
  <BrowserRouter>
    <div>
      <Header />

      <div>
        <Match
          exactly
          pattern="/"
          render={props => (
            <LazyRoute
              {...props}
              component={System.import('../../pages/Home')}
            />
          )}
        />

        <Match
          exactly
          pattern="/about"
          render={props => (
            <LazyRoute
              {...props}
              component={System.import('../../pages/About')}
            />
          )}
        />

        <Miss
          render={props => (
            <LazyRoute
              {...props}
              component={System.import('../../pages/HTTP404')}
            />
          )}
        />
      </div>
    </div>
  </BrowserRouter>
);

RootRouter.propTypes = {
};

// Jest doesn't like the System.import above,
// so we're just not really gonna test this for now.
if (process.env.NODE_ENV === 'test') {
  module.exports = () => <div>RootRouter</div>;
} else {
  module.exports = RootRouter;
}
