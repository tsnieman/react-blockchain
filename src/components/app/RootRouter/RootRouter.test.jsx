import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies
import { MemoryRouter } from 'react-router';
import RootRouter from './RootRouter';

test('RootRouter is tested', () => {
  const component = renderer.create(
    <MemoryRouter>
      <RootRouter />
    </MemoryRouter>,
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
