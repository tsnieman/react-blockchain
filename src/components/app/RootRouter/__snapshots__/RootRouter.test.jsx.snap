exports[`test RootRouter is tested 1`] = `
<div>
  RootRouter
</div>
`;
